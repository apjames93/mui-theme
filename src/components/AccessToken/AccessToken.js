import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Form, MuiInput, MuiSubmit } from 'rff-wrapper';
import {
  Tooltip,
  Dialog,
  DialogTitle,
  DialogContent,
  DialogActions,
  IconButton,
} from '@material-ui/core';
import TextFormatIcon from '@material-ui/icons/TextFormat';

export const AccessToken = ({ setAccessToken }) => {
  const [open, setOpen] = useState(false);

  const handleOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const onSubmit = async (values) => {
    setAccessToken({ accessToken: values.accessToken });
    handleClose();
  };

  return (
    <>
      <Tooltip title="Storyblok Access Token" placement="top-end">
        <IconButton onClick={handleOpen}>
          <TextFormatIcon />
        </IconButton>
      </Tooltip>
      <Dialog open={open} onClose={handleClose}>
        <DialogTitle>Add Storyblok Access Token</DialogTitle>
        <DialogContent>
          <Form onSubmit={onSubmit}>
            <MuiInput name="accessToken" placeholder="Storyblok Access Token" fullWidth />
            <DialogActions>
              <MuiSubmit buttonText="Use Token" />
            </DialogActions>
          </Form>
        </DialogContent>
      </Dialog>
    </>
  );
};

export default AccessToken;

AccessToken.propTypes = {
  setAccessToken: PropTypes.func,
};

AccessToken.defaultProps = {
  setAccessToken: undefined,
};
