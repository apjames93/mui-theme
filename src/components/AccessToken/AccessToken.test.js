import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import AccessToken from './AccessToken';

function setup() {
  const props = {
    setState: jest.fn(),
    theme: { primary: 'asdf' },
    accessToken: '1234',
    pageRedirect: '/',
  };

  const comp = mount(<AccessToken {...props} />);
  return { comp, props };
}

describe('<AccessToken />', () => {
  it('renders AccessToken', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  it('should handleSubmit and add Link tag to head', async () => {
    const { comp } = setup();
    const iconBtn = comp.find('ForwardRef(IconButton)').first();
    window.location.host = 'http://localhost:3000';

    iconBtn.simulate('click');

    const dialogOpen = comp.find('WithStyles(ForwardRef(Dialog))').first().prop('open');
    expect(dialogOpen).toEqual(true);

    // console.log(comp.find('input').at(1).props());
    const accessInput = comp.find('input').first();
    const pageInput = comp.find('input').at(1);
    accessInput.simulate('change', { target: { value: '9090' } });
    pageInput.simulate('change', { target: { value: '/kale' } });

    const submitBtn = comp.find('MuiSubmit').first();
    await submitBtn.simulate('submit');

    setTimeout(() => {
      expect(comp.find('AccessToken').first().prop('accessToken')).toEqual('9090');
    }, 400);
  });

  it('should handle close and close dialog when submitted or close is clicked', () => {
    const { comp } = setup();
    const dialog = comp.find('WithStyles(ForwardRef(Dialog))').first().props();
    expect(dialog.open).toEqual(false);
    const iconBtn = comp.find('ForwardRef(IconButton)').first();
    iconBtn.simulate('click');
    const dialogOpen = comp.find('WithStyles(ForwardRef(Dialog))').first().props();
    expect(dialogOpen.open).toEqual(true);
    const dialogClose = comp.find('WithStyles(ForwardRef(Dialog))').first().prop('onClose');
    act(() => {
      dialogClose();
    });
    setTimeout(() => {
      const dialogClosed = comp.find('WithStyles(ForwardRef(Dialog))').first().prop('open');
      expect(dialogClosed).toEqual(false);
    }, 800);
  });

  test('snapshot', () => {
    const tree = renderer.create(<AccessToken />);
    expect(tree).toMatchSnapshot();
  });
});
