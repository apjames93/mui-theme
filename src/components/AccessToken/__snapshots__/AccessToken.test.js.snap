// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccessToken /> snapshot 1`] = `
<button
  aria-describedby={null}
  className="MuiButtonBase-root MuiIconButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title="Storyblok Access Token"
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M5 17v2h14v-2H5zm4.5-4.2h5l.9 2.2h2.1L12.75 4h-1.5L6.5 15h2.1l.9-2.2zM12 5.98L13.87 11h-3.74L12 5.98z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
